### ISSUE TITLE
WMTS with 2154 projection

### ISSUE THREAD

**Issue by bmarsaud** [iTowns/itowns#1652]
## Your Environment

* Version used: 2.32.0
* Browser Name and version: Chrome 81
* Operating System and version (desktop or mobile): Ubuntu 20.04

## Context

I'm trying to display the `GEOGRAPHICALGRIDSYSTEMS.MAPS.SCAN25TOPO.L93` basemap layer from IGN via WMTS on a `PlanarView` with `EPSG:2154` projection.

## Steps to Reproduce (for bugs)

Here is a Gist with what I have done so far : https://gist.github.com/bmarsaud/95bc0bc53b056f6db75820cb3652a7a7

Please replace the `<key>` in the WMTS layer URL with a valid IGN key that have access to the Scan25Topo layer.

For testing purpose here is another 2154 WMTS in opendata : https://gist.github.com/bmarsaud/824687716f90a77a6d1784851877cbd5

## Expected Behavior

The basemap layer should be displayed on the map.

## Actual Behavior

The layer doesn't show up in iTowns.

When the `tileMatrixSetLimits` is set according to the WMTS `GetCapabilities`, no requests are sent to the service.

When the `tileMatrixSetLimits` is NOT set, some requests are sent but all return no data : tile indexes doesn't seem right.

## Possible Cause/Fix/Solution

* Bad projection declaration or extent ?

**Comment by gchoqueux** [comment]
The issue is the extent and it's difficult to calculate.
you can't find it directly in the `getCapabilities`
but there's `getCapabilities` constants to calculate it.

It would be interesting to add an automatic calculation tool in iTowns

```js
// WMTS OGC standard (meter)
const sizePixel = 0.00028;

// Info WMTS get capabalities
const scaleDenominator = 373497230.5353357571875677;
const tileSize = 256;
const TopLeftCorner = new itowns.Coordinates('EPSG:2154', 0, 12000000, 0);

// Compute the global extent
const resolution = sizePixel * scaleDenominator;
const dimension = tileSize * resolution;

const l93Extent = new itowns.Extent('EPSG:2154',
  TopLeftCorner.x,
  TopLeftCorner.x + dimension,
  TopLeftCorner.y - dimension,
  TopLeftCorner.y );

var mapElem = document.getElementById('map');
var view = new itowns.PlanarView(mapElem, l93Extent, {
    sseSubdivisionThreshold: 0.5,
    maxSubdivisionLevel: 21,
    placement: {
      coord: new itowns.Coordinates('EPSG:2154', 700000, 6600000, 0),
      range: 3000000,
    }
});
```

**Comment by bmarsaud** [comment]
Thanks for your answer!

If I understand correctly, the view is adapting to the WMTS. It's so not possible to have multiple WMTS with different tile matrices in the same view?
What about an MVT layer with 2154 data? As it's also tiled, would it work with this kind of extent change?

Also, I have tried to add GeoJSON data on top of this WMTS and the data is a bit off:
<details>
  <summary>Incorrect data in iTowns</summary>

![image](https://user-images.githubusercontent.com/2720587/117468286-d32d3100-af54-11eb-878c-ff3087c9dbed.png)
</details>

<details>
  <summary>Correct data in openlayers</summary>

![image](https://user-images.githubusercontent.com/2720587/117471016-b0e8e280-af57-11eb-8047-f0f54f68c619.png)
</details>

Do you have an idea to fix that? I was thinking about a problem with floating point precision on the scale denominator maybe. Like #1596.

**Comment by gchoqueux** [comment]
> If I understand correctly, the view is adapting to the WMTS. It's so not possible to have multiple WMTS with different tile matrices in the same view?

Yes, it's not possible for the moment. We want to add this feature. 
For the moment the projection coordinates are pre-calculate and stored in tiles vertex.
We want to replace it by projection matrices that will allow dynamic calculation of image positioning.
It will then be possible to overlay the images with different projections.

> What about an MVT layer with 2154 data? As it's also tiled, would it work with this kind of extent change?

If MVT projection data is EPSG 2154, it will work. Otherwise you will have to wait for the system with the projection matrices.

> Also, I have tried to add GeoJSON data on top of this WMTS and the data is a bit off:

I don't think it's the same issue as  #1596. (this issue will be resolved in the release 2.33.0)
but this `WMTS tileMatrixSet` is special. It doesn't have a regular subdivision,
there may be a loss of precision with the subdivision.

Could you up-load the geojson file, please?

**Comment by bmarsaud** [comment]
Ok, thanks for the clarification.

Here is the geojson: https://gist.github.com/bmarsaud/95bc0bc53b056f6db75820cb3652a7a7#file-chemins-de-randonnee-2154-json
